src/main/java/com/semmle/jira/addon/util/JiraUtils.java

```diff
@@ -43,16 +51,41 @@ public static IssueType getIssueTypeByName(String issueTypeName) {
                 ConstantsManager.CONSTANT_TYPE.ISSUE_TYPE.getType(), issueTypeName);
   }
 
+  public static IssueType getLgtmIssueType() {
+    return getIssueTypeByName(Constants.ISSUE_TYPE_NAME);
+  }
+
   public static void addWorkflowToProject(
-      Project project, JiraWorkflow workflow, IssueType issueType) throws GenericEntityException {
+      Project project, JiraWorkflow workflow, IssueType issueType, ApplicationUser user)
+      throws GenericEntityException, UsedIssueTypeException {
     WorkflowSchemeManager workflowSchemeManager = ComponentAccessor.getWorkflowSchemeManager();
     GenericValue workflowScheme = workflowSchemeManager.getWorkflowScheme(project);
-    workflowSchemeManager.addWorkflowToScheme(
-        workflowScheme, workflow.getName(), issueType.getId());
-  }
+    Map<String, String> workflowMap = workflowSchemeManager.getWorkflowMap(project);
+    if (workflowMap.containsKey(issueType.getId())) {
+      if (!workflowMap.get(issueType.getId()).equals(workflow.getName())) {
+        // Issue type already associated with another workflow
+        throw new UsedIssueTypeException();
+      }
+    } else {
+      JqlQueryBuilder builder = JqlQueryBuilder.newBuilder();
+      builder.where().project(project.getId()).and().issueType(issueType.getName());
+      try {
+        SearchResults results =
+            ComponentAccessor.getComponent(SearchService.class)
+                .searchOverrideSecurity(
+                    user, builder.buildQuery(), PagerFilter.getUnlimitedFilter());
```

Comment:
A `limited` pagerfilter might be better, we only need to know if any result exists.